File: Python/day06.py
def pass_one_day(aquarium: list):
    new_aquarium = [0, 0, 0, 0, 0, 0, 0, 0, 0]
    for i in range(8):
        new_aquarium[i] += aquarium[i+1]
    new_aquarium[8] += aquarium[0]
    new_aquarium[6] += aquarium[0]
    return new_aquarium

def read_input():
    aquarium = [0, 0, 0, 0, 0, 0, 0, 0, 0]
    with open('inputs/day06.txt', 'r') as file:
        list = file.read().split(",")
        for num in list:
            aquarium[int(num)] += 1
            
    return aquarium

def count_fish(aquarium: list):
    count = 0
    for nb in aquarium:
        count += nb
    return str(count)

def main():
    aquarium = read_input()
    print(aquarium)
    for i in range(256):
        aquarium = pass_one_day(aquarium)
    print("Ans : " + count_fish(aquarium))

main()

File: Go/day06.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func pass_on_day(aquarium [9]int) [9]int {
	new_aquarium := [9]int{0, 0, 0, 0, 0, 0, 0, 0, 0}
	for i := 0; i < 8; i++ {
		new_aquarium[i] += aquarium[i+1]
	}
	new_aquarium[8] += aquarium[0]
	new_aquarium[6] += aquarium[0]
	return new_aquarium
}

func read_input() [9]int {
	aquarium := [9]int{0, 0, 0, 0, 0, 0, 0, 0, 0}
	input_string, _ := os.ReadFile("inputs/day06.txt")
	input := strings.Split(string(input_string), ",")
	for _, num := range input {
		index, _ := strconv.Atoi(num)
		aquarium[index] += 1
	}
	return aquarium
}

func count_fish(aquarium [9]int) int {
	count := 0
	for _, nb := range aquarium {
		count += nb
	}
	return count
}

func main() {
	aquarium := read_input()
	for i := 0; i < 256; i++ {
		aquarium = pass_on_day(aquarium)
	}
	fmt.Printf("Ans : %d", count_fish(aquarium))
}
